src/data_loader.py

```diff
@@ -114,6 +117,9 @@ def load_data(self, file_path_list):
                     self.log.exception(msg)
                     self.errors.append(msg)
                     return False, self.errors
+            finally:
+                deleted_nodes = None
+                deleted_file_nodes = None
```

Comment:
After this "load_data" function, deleted_nodes, deleted_file_nodes will be de-allocated from the heap memory. I think we dont need to set None. Could you double check?

Reply:
explicitly set to None to make memory allocated immediately.